src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LIST_SIZE = 10;
+    private int size;
+    private Object[] list;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_LIST_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            list = increase(list);
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == list.length) {
+            list = increase(list);
+        }
+        if ((index >= 0) && (index <= size)) {
+            Object[] temporary = new Object[list.length];
+            System.arraycopy(list, 0, temporary,0, index);
+            temporary[index] = value;
+            System.arraycopy(list, index, temporary,index + 1, size - index);
+            list = temporary;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Your index is over the size");
```

Comment:
Le's make exception messages more informative. It is a good practice to add info about the size and the wrong index